stairway/src/main/java/bio/terra/stairway/impl/StairwayThreadPool.java

```diff
@@ -3,34 +3,15 @@
 import bio.terra.stairway.FlightContext;
 import java.util.Map;
 import java.util.concurrent.Future;
-import java.util.concurrent.LinkedBlockingQueue;
-import java.util.concurrent.ThreadPoolExecutor;
-import java.util.concurrent.TimeUnit;
-import java.util.concurrent.atomic.AtomicInteger;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 import org.slf4j.MDC;
+import org.springframework.scheduling.concurrent.ThreadPoolTaskExecutor;
 
-class StairwayThreadPool extends ThreadPoolExecutor {
-  private static final Logger logger = LoggerFactory.getLogger(StairwayThreadPool.class);
-  AtomicInteger activeTasks;
+class StairwayThreadPool {
 
-  StairwayThreadPool(int maxParallelFlights) {
-    super(
-        maxParallelFlights,
-        maxParallelFlights,
-        0L,
-        TimeUnit.MILLISECONDS,
-        new LinkedBlockingQueue<>());
-    activeTasks = new AtomicInteger();
-  }
-
-  int getActiveFlights() {
-    return activeTasks.get();
-  }
+  private final ThreadPoolTaskExecutor executor;
 
-  int getQueuedFlights() {
-    return getQueue().size();
+  StairwayThreadPool(ThreadPoolTaskExecutor executor) {
```

Comment:
I like the idea of moving the thread pool to a field rather than a base class. Is there a reason why you didn't continue to use `ThreadPoolExecutor` here? That way a user doesn't need to use a spring type to provide a custom thread pool. I believe the spring types are compatible with the java ones, so using `ThreadPoolExecutor` would still allow clients to use `ThreadPoolTaskExecutor` if they wanted to.

Reply:
Yes, that's the benefit that tipped the scale towards my usage of it over `ThreadPoolExecutor` (I liked that I was able to get rid of that custom tracking in this implementation as a result).  Tough to thread the needle sometimes between the diversity in tech stacks already present in our apps vs. what has become the prevailing standard (Spring Boot Java).

`ThreadPoolTaskExecutor` doesn't actually extend `ThreadPoolExecutor` either, so it would have been more involved (but not impossible) to support the non-Spring type.

Reply:
That's a good question, but as `StairwayThreadPool`s are not public, and we don't support their direct implementation by consumers.

I don't see this as a breaking change as far as our contracts with consumers are concerned, but rather an implementation detail, so I'm not inclined to maintain that old constructor.

Reply:
OMG you're totally right.  Sorry, been looking at python too long and missed the absence of the `public` qualifier.

Reply:
No worries at all, I'm glad for the sanity check!